src/core/rbuscore.c

```diff
@@ -1924,6 +1925,7 @@ rbusCoreError_t rbus_discoverObjectElements(const char * object, int * count, ch
                         for (int j = 0; j < i; j++)
                             free(array_ptr[j]);
                         free(array_ptr);
+                        array_ptr=NULL;
```

Comment:
Dont we have to set ```*elements = NULL;```  here?

Reply:
since *elements = array_ptr; was set earlier *element is pointing to array_ptr ; setting array_ptr to NULL should make *element point to NULL, 
If wrong please correct me

Reply:
Added prints and checking to confirm